src/sympc/tensor/replicatedshare_tensor.py

```diff
@@ -41,17 +44,55 @@ class ReplicatedSharedTensor(metaclass=SyMPCTensor):
     def __init__(
         self,
         shares: Optional[List[Union[float, int, torch.Tensor]]] = None,
-        session: Optional[Session] = None,
+        config: Config = Config(encoder_base=2, encoder_precision=16),
+        session_uuid: Optional[UUID] = None,
+        ring_size: int = 2 ** 64,
     ):
-        """Initialize ShareTensor.
+        """Initialize ReplicatedSharedTensor.
 
         Args:
             shares (Optional[List[Union[float, int, torch.Tensor]]]): Shares list
                 from which RSTensor is created.
-            session (Optional[Session]): The session. Defaults to None.
+            config (Config): The configuration where we keep the encoder precision and base.
+            session_uuid (Optional[UUID]): Used to keep track of a share that is associated with a
+                remote session
+            ring_size (int): field used for the operations applied on the shares
+                Defaults to 2**64
+
+        """
+        self.session_uuid = session_uuid
+        self.ring_size = ring_size
+
+        self.config = config
+        self.fp_encoder = FixedPointEncoder(
+            base=config.encoder_base, precision=config.encoder_precision
+        )
+
+        tensor_type = get_type_from_ring(ring_size)
+        self.shares = []
+        if shares is not None:
+            for i in range(len(shares)):
+                self.shares.append(self._encode(shares[i]).to(tensor_type))
+
+    def _encode(self, data):
+        return self.fp_encoder.encode(data)
+
+    def decode(self):
+        """Decode via FixedPrecisionEncoder.
+
+        Returns:
+            torch.Tensor: Decoded value
         """
-        self.session = session
-        self.shares = shares
+        return self._decode()
+
+    def _decode(self):
+
+        shares = []
+        for i in range(len(self.shares)):
+            tensor = self.fp_encoder.decode(self.shares[i].type(torch.LongTensor))
```

Comment:
Q: We convert it to long also in `ShareTensor`?

Reply:
Yeah ,George ,the semantics is that ,during decoding in the dividend [step](/home/rasswanth/.local/share/virtualenvs/python39-xmdK_WSC/lib/python3.9) ,the scale is cast to the tensor type for floor division. 

For Ex: if tensor_type is` torch.int16` and the scale is `2**17` , casting scale to 16 bit would result in zero which leads to `ZeroDivisonError`. So we cast to `torch.LongTensor` to accomodate large scale values.